queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java

```diff
@@ -128,31 +136,100 @@ private static void init() {
 
     static void connect() {
         try {
-            ConnectionFactory connectionFactory = new ConnectionFactory();
-            connectionFactory.setAutomaticRecoveryEnabled(true);
-            connectionFactory.setTopologyRecoveryEnabled(true);
 
+            final Map<String, Map<String, String>> serverConnections = ConfigUtil.parseComplexArrayStructure("service.rabbitmq.connections");
+            final ArrayList<String> serverConnectionKeys = Lists.newArrayList(serverConnections.keySet());
+            Collections.sort(serverConnectionKeys);
 
-            Configuration configuration = ConfigurationFactory.getConfiguration();
-            Configuration subsetBase = configuration.subset("service.rabbitmq");
-            Configuration subsetSecurity = subsetBase.subset("security");
+            Config config = new Config()
+                    .withRecoveryPolicy(new RecoveryPolicy()
+                            .withBackoff(Duration.seconds(1), Duration.seconds(30))
+                            .withMaxAttempts(20))
```

Comment:
please review last commit and let me know if it's ok. 1000 as default with interval of 30 seconds is very high, and high enough IMHO (more than 8 hours)!!!